UPGRADE_from_v5_to_v6.md

```diff
@@ -0,0 +1,207 @@
+Additionally, the `mender-x.y` tag has been replaced with the `vx.y` tag.
+* Enterprise registry:
+  From `registry.mender.io/mendersoftware/deployments-enterprise:mender-3.7`
+  to `registry.mender.io/mender-server-enterprise/deployments:v4.0`
+
+* Open Source registry:
+  From `docker.io/mendersoftware/deployments:mender-3.7`
+  to `docker.io/mendersoftware/deployments:v4.0`
+
+The new repository structure is already reflected in the default `values.yaml`
+file; please make sure to not override it in your `mender-values.yaml` file.
+
+### Drop the "mender-" prefix in the tags names
+The new image tags drop the `mender-` prefix, so this is no longer valid:
+
+```yaml
+# previous version no longer valid
+global:
+  image:
+    tag: mender-3.7.7
+```
+Instead the new tag is simply `v4.0.0`:
+
+```yaml
+# new version
+default:
+  image:
+    tag: v4.0.0
+```
+The default values is handling this already, so you don't need to change it.
+
+```yaml
+global:
+#  image:
+
+default:
+#  image:
+```
+
+
+### ImagePullSecrets instead cleartext credentials:
+**Enterprise version only**: for improved security, the `global.image.username`
+and `global.image.password` are deprecated in favor of `default:imagePullSecrets`:
+with this new Helm Chart release, you have to manually create a new Docker
+Registry secret, like this:
+```bash
+export MENDER_REGISTRY_USERNAME="replace-with-your-username"
+export MENDER_REGISTRY_PASSWORD="replace-with-your-password"
+export MENDER_REGISTRY_EMAIL="replace-with-your-email"
+
+kubectl create secret docker-registry my-mender-pull-secret \
+  --docker-username=${MENDER_REGISTRY_USERNAME} \
+  --docker-password=${MENDER_REGISTRY_PASSWORD} \
+  --docker-email=${MENDER_REGISTRY_EMAIL} \
+  --docker-server=registry.mender.io
+```
+and reference it in the `mender-values.yaml` file:
+
+```yaml
+global:
+  # image:
+  #   username: "redacted"
+  #   password: "redacted"
+default:
+  imagePullSecrets:
+    - name: my-mender-pull-secret
+```
+
+### Rootless gui container
+For improved security, the `gui` container is now rootles; this means that
+the `gui.httpPort` is switched from `80` to the unprivileged `8090` port.
+Make sure you are not overriding the `gui.httpPort` in your
+`mender-values.yaml` file.
+
+### Service Keys automatically generated
+You can choose to specify the `device_auth.certs.key`,
+`useradm.certs.key`, and `tenantadm.certs.key` keys in
+the `mender-values.yaml` file, but it is not mandatory anymore.
+If you don't specify them, the Helm Chart will generate them for you.
+
+### NATS and MongoDB subcharts enabled by default
+To facilitate the first Helm Chart installation, the MongoDB and NATS
+subcharts are enabled by default:
+```yaml
+mongodb:
+  enabled: true
+
+nats:
+  enabled: true
+```
+If you are using an external MongoDB or NATS instance, please make sure to
+explicitly disable them before the upgrade:
+```yaml
+mongodb:
+  enabled: false
+
+nats:
+  enabled: false
+```
```

Comment:
In this case, the default value is for the enterprise, which still have to change it (adapt storage, and Mongo).. Then the upcoming new users should wget/clone the repo, to grab the values-opensource.yaml, and run:
```
helm install mender mender/mender -f my-custom.yaml -f values-opensource.yaml
```
where the default `values.yaml` file for enterprise still applies.. This would overcomplicate i imho..